Freeing an array using the address of its second element is bad.

--- krb5-1.2.1/src/appl/gssftp/ftp/cmds.c.heap	Tue Sep 12 11:48:13 2000
+++ krb5-1.2.1/src/appl/gssftp/ftp/cmds.c	Tue Sep 12 11:48:43 2000
@@ -1843,6 +1843,7 @@
 	char **cpp;
 {
 	char **globbed;
+	char **globbed1;
 
 	if (!doglob)
 		return (1);
@@ -1856,10 +1857,11 @@
 		return (0);
 	}
 	if (globbed) {
-		*cpp = *globbed++;
+		globbed1 = globbed;
+		*cpp = *globbed1++;
 		/* don't waste too much memory */
 		if (*globbed) {
-			blkfree(globbed);
+			blkfree(globbed1);
 			free((char *)globbed);
 		}
 	}
